File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/_tokens/spacing/PaddingRoles.kt
@file:Suppress("ktlint:standard:package-name")

package com.perrystreet.woof.designsystem.atomic._tokens.spacing

import androidx.compose.ui.unit.Dp
import com.perrystreet.woof.designsystem.atomic._primitives.spacing.SpacingPrimitives

object PaddingRoles {
    enum class Screen(
        val dp: Dp,
    ) {
        ExtraCompact(SpacingPrimitives.Space4),
        Compact(SpacingPrimitives.Space8),
        Regular(SpacingPrimitives.Space20),
        Expanded(SpacingPrimitives.Space60),
    }

    enum class Element(
        val dp: Dp,
    ) {
        Compact(SpacingPrimitives.Space4),
        Regular(SpacingPrimitives.Space8),
        Relaxed(SpacingPrimitives.Space12),
        Expanded(SpacingPrimitives.Space20),
        ExtraExpanded(SpacingPrimitives.Space24),
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/_Tokens/Spacing/PaddingRoles.swift
import Foundation

public enum PaddingRoles {
    public enum Screen {
        case extraCompact
        case compact
        case regular
        case expanded

        public var rawValue: CGFloat {
            switch self {
            case .extraCompact: SpacingPrimitives.space4.rawValue
            case .compact: SpacingPrimitives.space8.rawValue
            case .regular: SpacingPrimitives.space20.rawValue
            case .expanded: SpacingPrimitives.space60.rawValue
            }
        }
    }

    public enum Element {
        case compact
        case regular
        case relaxed
        case expanded
        case extraExpanded

        public var rawValue: CGFloat {
            switch self {
            case .compact: SpacingPrimitives.space4.rawValue
            case .regular: SpacingPrimitives.space8.rawValue
            case .relaxed: SpacingPrimitives.space12.rawValue
            case .expanded: SpacingPrimitives.space20.rawValue
            case .extraExpanded: SpacingPrimitives.space24.rawValue
            }
        }
    }
}
